cmd/ocsp-updater/main.go

```diff
@@ -236,26 +236,56 @@ func (updater *OCSPUpdater) findStaleOCSPResponses(oldestLastUpdatedTime time.Ti
 	// the query below can be rewritten to use `AND NOT cs.isExpired`.
 	now := updater.clk.Now()
 	maxAgeCutoff := now.Add(-updater.ocspStaleMaxAge)
-	_, err := updater.dbMap.Select(
-		&statuses,
-		`SELECT
-			 cs.serial,
-			 cs.status,
-			 cs.revokedDate
-			 FROM certificateStatus AS cs
-			 JOIN certificates AS cert
-			 ON cs.serial = cert.serial
-			 WHERE cs.ocspLastUpdated > :maxAge
-			 AND cs.ocspLastUpdated < :lastUpdate
-			 AND cert.expires > now()
-			 ORDER BY cs.ocspLastUpdated ASC
-			 LIMIT :limit`,
-		map[string]interface{}{
-			"lastUpdate": oldestLastUpdatedTime,
-			"maxAge":     maxAgeCutoff,
-			"limit":      batchSize,
-		},
-	)
+
+	// If CertStatusOptimizationsMigrated is enabled then we can do this query
+	// using only the `certificateStatus` table, saving an expensive JOIN and
+	// improving performance substantially
+	var err error
+	if features.Enabled(features.CertStatusOptimizationsMigrated) {
+		_, err = updater.dbMap.Select(
```

Comment:
Since the query string is the bulk of the code here, I'd rather have the simplicity of just duplicating it. That makes it simpler to delete one of the if branches on the feature flag is flipped.